lib/rules/no-log.js

```diff
@@ -2,31 +2,37 @@
 
 const Rule = require('./base');
 
-const message = 'Unexpected {{log}} usage.';
+const ERROR_MESSAGE = 'Unexpected {{log}} usage.';
 
 module.exports = class NoLog extends Rule {
-  _checkForLog(node) {
-    if (node.path.original === 'log') {
-      this.log({
-        message,
-        line: node.loc && node.loc.start.line,
-        column: node.loc && node.loc.start.column,
-        source: this.sourceForNode(node),
-      });
-    }
-  }
-
   visitor() {
     return {
-      MustacheStatement(node) {
-        this._checkForLog(node);
+      BlockStatement(node) {
+        if (this.isLogNode(node)) {
```

Comment:
I think both `BlockStatement` and `MustacheStatement` need to do the `!this.isLocal(node)` guard. 

Additionally (though this isn't this rules job) `{{#log` is **never** going to be the built-in log helper AFAIK.

Reply:
Updated. I was changing the existing code a bit too much to my personal preference. I reverted this. The diff should be cleaner now.